### PR TITLE
(chore) Update npm dependencies and remove deprecared packages

### PR DESCRIPTION
**Pull request by abadri**
(chore) Update npm dependencies and remove deprecared packages
1. Upgrade karma, qunit
2. Migrate qunit tests from v1 to v2 using codemod
3. Replace esperanto with rollup

This fixes https://github.com/pretenderjs/FakeXMLHttpRequest/issues/57 

Note: Not a major version bump as all are build time dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,23 +6,28 @@
...
+    "karma-firefox-launcher": "^2.1.0",
     "karma-phantomjs-launcher": "^1.0.4",
-    "karma-qunit": "^1.2.1",
-    "qunitjs": "^1.0.0"
+    "karma-qunit": "^4.1.2",
+    "qunit": "^2.14.0",
+    "rollup": "^2.40.0"
+  },
+  "dependencies": {
+    "qunit-migrate": "^2.0.2"
```
Comment (xg-wang) [inline package.json:31]: Should this be removed?

File: package.json [hunk 2]
```diff
@@ -6,23 +6,28 @@
...
   "repository": "https://github.com/trek/FakeXMLHttpRequest.git",
   "scripts": {
-    "test": "esperanto -i src/fake-xml-http-request.js -t umd -b -n FakeXMLHttpRequest -o fake_xml_http_request.js && ./node_modules/karma/bin/karma start --single-run --browsers PhantomJS"
+    "start": "karma start",
+    "build": "rollup src/fake-xml-http-request.js --file fake_xml_http_request.js --format umd --name 'FakeXMLHttpRequest' --sourcemap true",
+    "test": "npm run build && karma start --single-run --browsers PhantomJS"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=14"
```
Comment (xg-wang) [inline package.json:14]: We should remove engines.node if not used in node at all, otherwise yarn/npm would complain in dependents.

**Review by xg-wang — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (abadri) [reply package.json]: Oops thanks for catching.

File: package.json (hunk 2 above)
Comment (abadri) [reply package.json]: Removed.

**Review by xg-wang — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]

**Comment by xg-wang** [comment]
@abadri Thank you!

**Pull request merged**